src/provider.tsx

```diff
@@ -1,35 +1,49 @@
-import React, { createContext, useContext } from "react";
+import { createContext, PropsWithChildren, useContext, useRef } from "react";
+import { HttpClient } from "./http";
 
 interface DuneContextType {
-  duneApiKey: string;
+  client: HttpClient;
 }
 
-const DuneContext = createContext<DuneContextType>({duneApiKey: ""})
+const DuneContext = createContext<DuneContextType | undefined>(undefined);
```

Comment:
why now allow undefined?

Reply:
Because `duneApiKey: ""` does not really make sense 😅  In the current code this would be allowed, but fail in the API request without telling the user how to solve it. Now instead the `useDuneContext` hook will fail early and let the user know what's wrong (missing DuneProvider).